packages/core-app-api/src/apis/system/ApiProvider.tsx

```diff
@@ -53,6 +53,24 @@ export const ApiProvider = (props: PropsWithChildren<ApiProviderProps>) => {
   );
 };
 
+/**
+ * Same as above, but does not inherit APIs from API providers further up in
+ * the react tree.
+ *
+ * Private to (and only used in) this package.
+ */
+export const IsolatedApiProvider = (
```

Comment:
Another possibility: adding an `isolated` prop to `<ApiProvider />` and letting anyone take advantage of it.

...I don't know how useful it is, and I kind of like being able to do this without affecting any APIs.

Reply:
I'd also say to keep it in here until the need arises in more places.